python/tvm/relay/frontend/pytorch.py

```diff
@@ -0,0 +1,1023 @@
+                if any(inp.debugName() in self._parsed_node_names.keys() \
+                       for inp in op_node.inputs()):
+                    listconstr = []
+                    for i in op_node.inputs():
+                        if i.debugName() in self._parsed_node_names.keys():
+                            listconstr.append( \
+                                outputs[self._parsed_node_names[i.debugName()]])
+                        elif i.node().kind() == "prim::Constant":
+                            listconstr.append(int(self._consts[i.debugName()]))
+                        elif i.debugName() in self._inputs_r.keys():
+                            listconstr.append(int(self._inputs_r[i.debugName()]))
+
+                    # Unwrap for tensors
+                    if len(listconstr) == 1:
+                        listconstr = listconstr[0]
+
+                    outputs.append(listconstr)
+                    self._parsed_node_names[op_name] = nid
+                    nid = nid+1
+            elif op_node.kind() != "prim::Constant":
+                for i in op_node.inputs():
+                    if i.debugName() in self._parsed_node_names.keys():
+                        for cnt in range(0, len(self._op_inputs_r[op_name])):
+                            if isinstance(self._op_inputs_r[op_name][cnt], str):
+                                if "call/var" in self._op_inputs_r[op_name][cnt]:
+                                    self._op_inputs_r[op_name][cnt] = \
+                                        outputs[self._parsed_node_names[i.debugName()]]
+                                    break
+
+                call = _convert_map[op_node.kind()](self._op_inputs_r[op_name],
+                                                    self._op_inputs_types[op_name])
+
+                outputs.append(call)
+                self._parsed_node_names[op_name] = nid
+                nid = nid+1
+
+        func = tvm.relay.Function(_analysis.free_vars(outputs[-1]), outputs[-1])
+
+        param = {k: tvm.nd.array(v) for k, v in self._param_tensors.items()}
+
+        return  _module.Module.from_expr(func), param
+
+    def _parse_inputs(self):
+        """ Map inputs to parser and inputs to graph. """
+        # Get names and objects of inputs for IR
+        ir_inputs = [i for i in self._graph.inputs()]
+
+        # Create corresponding shape and add to input
+        for input_name, ir_input in zip(self._input_shapes, ir_inputs[1:]):
+            input_shape = self._input_shapes[input_name]
+            ir_input.setDebugName(input_name)
+
+            ir_dtype = _convert_data_type(ir_input.type().scalarType().lower())
+            self._inputs_r[input_name] = _expr.var(input_name,
+                                                   shape=self._input_shapes[input_name],
+                                                   dtype=ir_dtype)
+
+        # Add self (first input of a PyTorch graph) to inputs
+        input_shape = [3]
```

Comment:
Where is 3 comes from?

Reply:
This is a magic number. Truthfully, the value doesn't matter just the name which I'll add as a comment.